test/cpplitetests.cpp

```diff
@@ -0,0 +1,510 @@
+
+    errno = 0;
+    blobs = list_all_blobs(container_name, "/", "");
+    ASSERT_EQ(0, errno);
+    ASSERT_EQ(1, blobs.size());
+
+    std::string dest_path = tmp_dir + "/destfile";
+
+    time_t lmt;
+    test_blob_client_wrapper->download_blob_to_file(container_name, blob_1_name, dest_path, lmt);
+
+    std::string result_text;
+    read_from_file(dest_path, result_text);
+
+    ASSERT_EQ(0, file_text.compare(result_text)) << "Strings not equal.  file_text = " << file_text << ", result_text = " << result_text;
+}
+
+// Helper method to create random file.
+void write_random_data_to_file(std::string path, unsigned int seed, size_t count)
+{
+    std::ofstream file_stream(path, std::ios::binary);
+    // Using a low-quality RNG here for speed, no need for high-quality randomness
+    std::minstd_rand r(seed);
+
+    for (size_t i = 0; i < count; i += 4 /* sizeof uint_fast32_t */)
+    {
+        uint_fast32_t val = r();
+        file_stream.write(reinterpret_cast<char*>(&val), 4);
+    }
+}
+
+// Helper method to validate the contents of a file match what is expected from the RNG, with a given seed.
+void read_file_data_and_validate(std::string path, unsigned int seed, size_t count)
+{
+    std::ifstream file_stream(path, std::ios::ate | std::ios::binary);
+    size_t file_size = file_stream.tellg();
+    ASSERT_EQ(count, file_size) << "File size incorrect.  Expected = " << count << ", actual = " << file_size;
+    file_stream.seekg(0);
+
+    std::minstd_rand r(seed);
+        for (size_t i = 0; i < count; i += 4 /* sizeof uint_fast32_t */)
+    {
+        uint_fast32_t expect_val = r();
+        uint_fast32_t actual_val;
+        file_stream.read(reinterpret_cast<char*>(&actual_val), 4);
+        ASSERT_EQ(expect_val, actual_val) << "File data incorrect at position "<< i << ".  Expected = " << expect_val << ", actual = " << actual_val;
+    }
+}
+
+// Helper method to validate uploading and download a single file of arbitrary size
+void BlobClientWrapperTest::run_upload_download(size_t file_size)
+{
+    errno = 0;
+    std::vector<list_blobs_hierarchical_item> blobs = list_all_blobs(container_name, "/", "");
+    ASSERT_EQ(0, errno);
+    ASSERT_EQ(0, blobs.size());
+
+    // Create a file
+    std::string file_path = tmp_dir + "/tmpfile";
+
+    unsigned int seed = 17;
+    write_random_data_to_file(file_path, seed, file_size);
+
+    std::string blob_1_name("blob1name");
+    test_blob_client_wrapper->upload_file_to_blob(file_path, container_name, blob_1_name);
+
+    errno = 0;
+    blobs = list_all_blobs(container_name, "/", "");
+    ASSERT_EQ(0, errno);
+    ASSERT_EQ(1, blobs.size());
+    ASSERT_EQ(file_size, blobs[0].content_length) << "Blob found, but size incorrect.";
+
+    std::string dest_path = tmp_dir + "/destfile";
+
+    time_t lmt;
+    test_blob_client_wrapper->download_blob_to_file(container_name, blob_1_name, dest_path, lmt);
+    // TODO: test that download returns correct blob properties once implemented.
```

Comment:
Test the lmt

Reply:
Good point, these tests were written before the LMT stuff was in, then rebased.  I'll add a new test just for LMT testing.

Reply:
Changed my mind, added checks in with the blob properties test